From 5f2839654aed9319d7ba19131ce9bf3788de83ab Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 25 Jun 2013 13:21:44 +0300
Subject: [PATCH 213/290] be: simplify `filler' parameter of m0_be_tx_init()

filler() does not need `datum' parameter, as it can be obtained from
tx->t_datum.
---
 be/tx.c | 2 +-
 be/tx.h | 5 ++---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/be/tx.c b/be/tx.c
index d08ce6e..774c4b6 100644
--- a/be/tx.c
+++ b/be/tx.c
@@ -178,7 +178,7 @@ M0_INTERNAL void m0_be_tx_init(struct m0_be_tx *tx, uint64_t tid,
 			       m0_be_tx_cb_t persistent,
 			       m0_be_tx_cb_t discarded,
 			       void (*filler)(struct m0_be_tx *tx,
-					      void *payload, void *datum))
+					      void *payload))
 {
 	m0_sm_init(&tx->t_sm, &tx_sm_conf, M0_BTS_INIT, tx_grp);
 	tx->t_root       = NULL;
diff --git a/be/tx.h b/be/tx.h
index 661325a..90ffe3f 100644
--- a/be/tx.h
+++ b/be/tx.h
@@ -441,8 +441,7 @@ struct m0_be_tx {
 	 * A typical use of this call-back is to form a "fol record" used by DTM
 	 * for distributed transaction management.
 	 */
-	void                 (*t_filler)(struct m0_be_tx *tx, void *payload,
-					 void *datum);
+	void                 (*t_filler)(struct m0_be_tx *tx, void *payload);
 	/**
 	 * User-specified value, associated with the transaction. Transaction
 	 * engine doesn't interpret this value. It can be used to pass
@@ -480,7 +479,7 @@ M0_INTERNAL void m0_be_tx_init(struct m0_be_tx *tx, uint64_t tid,
 			       m0_be_tx_cb_t persistent,
 			       m0_be_tx_cb_t discarded,
 			       void (*filler)(struct m0_be_tx *tx,
-					      void *payload, void *datum));
+					      void *payload));
 
 M0_INTERNAL void m0_be_tx_fini(struct m0_be_tx *tx);
 
-- 
1.8.3.2

